syntax = "proto3";

package rti;

message RithmicOrderNotification
	{       
	enum NotifyType {
    NOTIFYTYPE_UNSPECIFIED = 0;
	                 ORDER_RCVD_FROM_CLNT      =   1;
			 MODIFY_RCVD_FROM_CLNT     =   2;
			 CANCEL_RCVD_FROM_CLNT     =   3;
			 OPEN_PENDING              =   4;
			 MODIFY_PENDING            =   5;
			 CANCEL_PENDING            =   6;
			 ORDER_RCVD_BY_EXCH_GTWY   =   7;
			 MODIFY_RCVD_BY_EXCH_GTWY  =   8;
			 CANCEL_RCVD_BY_EXCH_GTWY  =   9;
			 ORDER_SENT_TO_EXCH        =  10;
			 MODIFY_SENT_TO_EXCH       =  11;
			 CANCEL_SENT_TO_EXCH       =  12;
			 OPEN                      =  13;
			 MODIFIED                  =  14;
			 COMPLETE                  =  15;
			 MODIFICATION_FAILED       =  16;
			 CANCELLATION_FAILED       =  17;
			 TRIGGER_PENDING           =  18;
			 GENERIC                   =  19;
			 LINK_ORDERS_FAILED        =  20;
                        }	

        enum TransactionType {
    TRANSACTIONTYPE_UNSPECIFIED = 0;
	                      BUY  = 1;
		              SELL = 2;
			      SS   = 3;
	                     }

        enum Duration {
    DURATION_UNSPECIFIED = 0;
	               DAY  = 1;
		       GTC  = 2;
		       IOC  = 3;
		       FOK  = 4;
                      }
 
        enum PriceType {
    PRICETYPE_UNSPECIFIED = 0;
	                LIMIT        = 1;
			MARKET       = 2;
			STOP_LIMIT   = 3;
			STOP_MARKET  = 4;
                       }

	enum BracketType {
    BRACKETTYPE_UNSPECIFIED = 0;
	                  STOP_ONLY               = 1;
			  TARGET_ONLY             = 2;
			  TARGET_AND_STOP         = 3;
			  STOP_ONLY_STATIC        = 4;
			  TARGET_ONLY_STATIC      = 5;
			  TARGET_AND_STOP_STATIC  = 6;
                         }

        enum OrderPlacement {
    ORDERPLACEMENT_UNSPECIFIED = 0;
	                     MANUAL  = 1;
			     AUTO    = 2;
	                    }

	int32             template_id            = 154467;
	string            user_tag               = 154119;

	NotifyType        notify_type            = 153625;
	bool              is_snapshot            = 110121;

	string            status                 = 110303;
	string            basket_id              = 110300;
	string            original_basket_id     = 154497;
	string            linked_basket_ids      = 110358;

	string            fcm_id                 = 154013;
	string            ib_id                  = 154014;
	string            user_id                = 131003;
	string            account_id             = 154008;

	string            symbol                 = 110100;
	string            exchange               = 110101;
	string            trade_exchange         = 112021;
	string            trade_route            = 112016;
	string            exchange_order_id      = 149238;
	string            instrument_type        = 110116;
	string            completion_reason      = 149273;

	 int32            quantity               = 112004;
	 int32            quan_release_pending   = 112027;
	 double           price                  = 110306;
	 double           trigger_price          = 149247;

	 TransactionType  transaction_type       = 112003;
	 Duration         duration               = 112005;
	 PriceType        price_type             = 112008;
	 PriceType        orig_price_type        = 154770;
	 OrderPlacement   manual_or_auto         = 154710;
	 BracketType      bracket_type           = 157087;

	 double           avg_fill_price         = 110322;

	 int32            total_fill_size        = 154111;
	 int32            total_unfilled_size    = 154112;

 	string            sequence_number        = 112002;
	string            orig_sequence_number   = 149263;
	string            cor_sequence_number    = 149264;

	string            currency               = 154382;
	string            country_code           = 154172;

	string            text                   = 120008;
	string            report_text            = 120028;
	string            remarks                = 154806;

	string            window_name            = 154629;
	string            originator_window_name = 154671;

	 int32            cancel_at_ssboe        = 157085;
	 int32            cancel_at_usecs        = 157086;
         int32            cancel_after_secs      = 154488;

	int32             ssboe                  = 150100;
	int32             usecs                  = 150101;
	}
